src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs

```diff
@@ -29,6 +32,7 @@ public static void TestResolveRelativePaths()
                 if (shouldDelete)
                 {
                     File.Delete(path);
+                    Directory.SetCurrentDirectory(curDir);
```

Comment:
I would put everything in the 'try'. As it is, if eg File.Open throws, the current directory is not reset.